section2/reflection.md

```diff
@@ -2,14 +2,29 @@
+
+
 4. What is a conditional statement? Give one example of a daily life conditional. Give one example of where a conditional is probably used in a web application you use.
 
+- A condition statement in life: Am I hungry? If yes, eat something. If no, don't eat. I could see a website applying conditionals when there is two versions of that website. The nytimes website, for example, provides a conditional to their paid and unpaid users: Are you a paid user? If yes, login to read the article. If no, pay us or leave.
+
 5. How do you add multiple conditions to an `if` statement? In your own words, explain how to program reads them and determines what to do.
 
+- Applying multiple conditions to an `if` statment will require several lines of code. It essentially works like a flow chart, where boolean data determines which direction the code will flow. Taking our real life example from the last question, we could write code that would ask `'Are you hungry?'` If the response was `yes`, or `true`, our first line of code would direct a script of `'Eat an oreo!'` to display. If the asnwer was `no`, then the condition of the first script would not apply, and therefore would not run. It would then move to our next line of code, written as an `if` statement prepared to respond in the case of `no`. With that input, we would design a task for this condition, to run the script `'Then don't eat!'`. 
+
 6. What tools are available to you, as a developer, if you want to check that TWO conditions are met? Or, if you want to check that 1 of 2 conditions are met, before running a block of code?
 
-7. What questions do you still have about `if` statements and/or functions?
\ No newline at end of file
+- To check this information, you can use logical operators. Using `&&` you can compare two conditions. The `&&` will only return a `true` if _both_ conditions are met. On the other hand, `||` will return `true` as long as _one_ of the conditions are met. In this way, you can check if two conditions are met or if 1/2 conditions are met before running a block of code. 
+
+7. What questions do you still have about `if` statements and/or functions?
+- I guess I'm continually wondering about how complicated these functions could become. Mentally I assume that the order of the `if` statements would start to matter if you had many `if` conditions running at once. I suppose I would like to know common mistakes programmers make when ordering more complicated programs. 
```

Comment:
Yeah great questions. There are absolutely times that the order matters. As you get into writing apps by mid-Mod 1, you'll start to see this come to life more, with a project that gives it more context.

RE mistakes - I don't even know if I'd call them mistakes when regarding the design of a program, but developers will have varying opinions, and there are usually many ways to solve a problem. In development, we do "code reviews" where a peer or supervisor reviews your code before finalizing it as part of the app. During those code reviews, others will look at your design choices and give feedback, have convos about it, etc. to determine ideal solutions.